src/parameters/define_parameter.ts

```diff
@@ -0,0 +1,15 @@
+import {
+  TypedObjectParameterDefinition,
+  TypedObjectProperties,
+  TypedObjectRequiredProperties,
+} from "../parameters/types.ts";
+
+export const DefineParameter = <
```

Comment:
The only purpose this method serves is to create a 'hook' for the type system to be able to apply the two generics below this line, and in particular, that is important because the generics sequentially rely on each other; that is, the third generic relies on the previous two named generics, and the second generic relies on the first named generic.